iceoryx_dds/include/iceoryx_dds/gateway/iox_to_dds.hpp

```diff
@@ -34,14 +35,28 @@ template <typename channel_t = gw::Channel<popo::UntypedSubscriber, dds::data_wr
 class Iceoryx2DDSGateway : public gateway_t
 {
   public:
+    /// @brief Default constructor
     Iceoryx2DDSGateway() noexcept;
+
+    /// @brief Load the provided gateway configuration
+    /// @param[in] config The gateway configuration to load
     void loadConfiguration(const config::GatewayConfig& config) noexcept;
+
+    /// @brief Discover messages coming from iceoryx.
+    /// @param[in] msg The discovery message.
     void discover(const capro::CaproMessage& msg) noexcept;
```

Comment:
What happens whet it fails to setup a channel?

Reply:
There's no error evaluation of `setupChannel()` here.